mirgecom/viscous.py

```diff
@@ -275,3 +279,37 @@ def get_viscous_cfl(discr, eos, dt, cv):
         The CFL at each node.
     """
     return dt / get_viscous_timestep(discr, eos=eos, cv=cv)
+
+
+def get_local_max_species_diffusivity(transport, eos, cv):
+    """Return the maximum species diffusivity at every point.
+
+    Parameters
+    ----------
+    transport: mirgecom.transport.TransportModel
+        A model representing thermo-diffusive transport
+    eos: mirgecom.eos.GasEOS
+        An equation of state implementing the speed_of_sound method
+    cv: :class:`~mirgecom.fluid.ConservedVars`
+        Fluid solution
+    """
+    actx = cv.array_context
+
+    if(transport is None or transport._d_alpha.size == 0):
+        return 0 * cv.mass / cv.mass
+
+    species_diffusivity = transport.species_diffusivity(eos, cv)
+    stacked_diffusivity = actx.np.stack([x[0] for x in species_diffusivity])
```

Comment:
Good catch. And if `x` is a DOFArray, you shouldn't need to do `x[0]` in the list comprehension I think.

Reply:
`transport.species_diffusivity` should return a tuple of `DOFArray`s. I was doing it this way because `DOFArray`s don't have some features that their underlying arrays do (like shape) but I can change it if you want.

Reply:
Yeah, `DOFArray`s won't have a single shape, because the per-mesh-group arrays inside them may have different shapes (corresponding to different element types/counts). We will want to make sure we're handling the multiple mesh group case here; this will probably mean calling the loopy kernel once per group.

Reply:
Ok, I looped over all possible mesh groups, calling the kernel once for each one. Unfortunately, if size can change, I have to make a new kernel each time as far as I know, but I don't think there's much overhead there.